chains/evm/contracts/libraries/ExtraArgsCodec.sol

```diff
@@ -0,0 +1,572 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+/// @notice Gas-optimized assembly version of ExtraArgsCodec library.
+library ExtraArgsCodec {
+  error InvalidDataLength(EncodingErrorLocation location);
+  error InvalidExecutorLength(uint256 length);
+  error InvalidCCVAddressLength(uint256 length);
+  error CCVArrayLengthMismatch(uint256 ccvsLength, uint256 ccvArgsLength);
+
+  bytes4 public constant GENERIC_EXTRA_ARGS_V3_TAG = 0x302326cb;
+  bytes4 public constant SVM_EXECUTOR_ARGS_V1_TAG = 0x1a2b3c4d;
+  bytes4 public constant SUI_EXECUTOR_ARGS_V1_TAG = 0x5e6f7a8b;
+
+  // Base size excludes all variable-length fields (CCV addresses/args, executor address, executorArgs, tokenReceiver,
+  // tokenArgs).
+  // Encoding order: tag(4) + gasLimit(4) + finalityConfig(2) + ccvsLength(1) + executorLength(1) +
+  // executorArgsLength(2) + tokenReceiverLength(2) + tokenArgsLength(2) = 19 bytes.
+  uint256 public constant GENERIC_EXTRA_ARGS_V3_BASE_SIZE = 4 + 4 + 2 + 1 + 1 + 2 + 2 + 2;
+  // Base size: tag(4) + useATA(1) + accountIsWritableBitmap(8) + accountsLength(1) = 14 bytes.
+  uint256 public constant SVM_EXECUTOR_ARGS_V1_BASE_SIZE = 4 + 1 + 8 + 1;
+  // Base size: tag(4) + objectIdsLength(1) = 5 bytes.
+  uint256 public constant SUI_EXECUTOR_ARGS_V1_BASE_SIZE = 4 + 1;
+
+  // Enum to indicate specific error locations during encoding/decoding.
+  enum EncodingErrorLocation {
```

Comment:
Is part of the intention here to prevent having to update this for new non-EVMs?

If so, how is space for params that new non-EVMs may need accounted for?